### PR TITLE
Refactor showPeople helper function

### PR DESCRIPTION
**Pull request by saschanaz**
This PR uses hyperHTML for showPeople function, replacing previous simple-but-hard-to-read string concatenations.

cc: @marcoscaceres 

Probably a good chance to do some chat and make things in right direction, before any migrations for core IDL templates.

### CODE REVIEW

**Comment by saschanaz** [comment]
I still have a concern about the comment nodes hyperHTML implicitly creates (https://github.com/WebReflection/hyperHTML/issues/194), what do you think?

File: src/w3c/headers.js [hunk 1]
```diff
@@ -109,137 +110,96 @@ const W3CDate = new Intl.DateTimeFormat(["en-AU"], {
...
-        bn = i < n - 1 ? "_:editor" + (i + 1) : "rdf:nil";
-        re = " resource='" + bn + "'";
-      }
-    } else {
-      ret += "<dd class='p-author h-card vcard'" + editorid + ">";
-    }
+  const attr = Object.freeze(self.doRDFa ? {
+    ...name === "Editor" ? {
+      rp1: "rdf:first",
+      rp2: "foaf:Person"
```
Comment (marcoscaceres) [inline src/w3c/headers.js]: Based on the discussion in the related bug, it seems safe to remove this stuff.

File: src/w3c/headers.js [hunk 2]
```diff
@@ -109,137 +110,96 @@ const W3CDate = new Intl.DateTimeFormat(["en-AU"], {
...
+    } : {},
+    rn: "foaf:name",
+    rm: "foaf:mbox",
+    rwu: "foaf:workplaceHomepage",
+    rpu: "foaf:homepage",
+    propSeeAlso: "rdfs:seeAlso"
+  } : {});
+  let ret = "";
+  const bns = [];
+  if (self.doRDFa && name === "Editor") {
```
Comment (marcoscaceres) [inline src/w3c/headers.js]: As above.

File: src/w3c/headers.js [hunk 3]
```diff
@@ -109,137 +110,96 @@ const W3CDate = new Intl.DateTimeFormat(["en-AU"], {
...
-          p.name +
-          "</a>";
-      } else
-        ret +=
-          "<a class='u-url url p-name fn' href='" +
-          p.url +
-          "'>" +
-          p.name +
-          "</a>";
+      if (self.doRDFa) {
```
Comment (marcoscaceres) [inline src/w3c/headers.js]: Same here

File: src/w3c/headers.js [hunk 4]
```diff
@@ -109,137 +110,96 @@ const W3CDate = new Intl.DateTimeFormat(["en-AU"], {
...
+  }
+
+  function getExtra(extra) {
+    const span = html`<span class='${extra.class || null}'></span>`
+    let textContainer = span;
+    if (extra.href) {
+      textContainer = html`<a
+        href='${extra.href || null}'
+        property='${self.doRDFa ? "rdfs:seeAlso" : null}'
+      ></a>`;
```
Comment (marcoscaceres) [inline src/w3c/headers.js]: Here too. Let’s get rid of all the property attributes.

File: src/w3c/headers.js (hunk 1 above)
Comment (saschanaz) [reply src/w3c/headers.js]: Oh, was that the conclusion? I thought https://github.com/w3c/respec/issues/1503#issuecomment-365561010 is saying that Google actively uses this.

Comment (marcoscaceres) [reply src/w3c/headers.js]: Yes and no :) Consider:

> "Whether the specific vocabulary and data patterns you're publishing map to a current public facing question is another matter."

Ideally, we should only include things that *do map* to "rich snippets". None of these seem to show up in any "rich snippets", so IMO they should go.

Comment (marcoscaceres) [reply src/w3c/headers.js]: I made a proposal in the bug: when refactoring code, if we find stuff that doesn't show up as a rich snippet, it should be removed from ReSpec (as it has little, to no, value). However, we should keep anything that does show up.

**Comment by saschanaz** [comment]
This was included in #1514 and now merged. We can discuss more in #1503 👍

**Pull request closed without merging**